### ISSUE TITLE
[routing] Cycling routing: handling of highway=pedestrian

### ISSUE THREAD

**Issue by patepelo** [organicmaps/organicmaps#7934]
In this case OM cycling routing is too inefficient (1.1km vs 500m) in comparison to other routing engines (Valhalla, OSRM)
![imagen](https://github.com/organicmaps/organicmaps/assets/23003127/60833d8d-a0a3-4252-a9e4-bddec6a9ce07)

Example:

https://www.openstreetmap.org/directions?engine=fossgis_valhalla_bicycle&route=41.40096%2C2.15985%3B41.39937%2C2.15735#map=19/41.40058/2.15842
![imagen](https://github.com/organicmaps/organicmaps/assets/23003127/5cbe5d3a-2a9e-4541-bec0-abc27f5c266f)

Looking into map data, it seems that highway=pedestrian will be avoided in OM (https://www.openstreetmap.org/way/936359817)

(latest beta version and map data)

@vng

**Comment by matkoniecz** [comment]
Is https://www.openstreetmap.org/way/936359817 allowing bicycle traffic?

If yes, then `bicycle=yes` is missing in OSM data.

If no, then avoiding it is entirely reasonable.

**Comment by patepelo** [comment]
> Is https://www.openstreetmap.org/way/936359817 allowing bicycle traffic?
> 
> If yes, then `bicycle=yes` is missing in OSM data.
> 
> If no, then avoiding it is entirely reasonable.


The [wiki:highway=pedestrian](https://wiki.openstreetmap.org/wiki/Tag:highway%3Dpedestrian) is not clear when not defined whether to assume bicycle yes or no in this case. 

I would set it up to yes when not defined. The cyclist would have the possibility to push in the case there is a sign explicitly prohibiting.

WDYT?

**Comment by Misalf-git** [comment]
Maybe it's area dependent but a pedestrian literally is a walking person.

In the english wiki I read "mainly or exclusively for pedestrians" ... "for pedestrians to walk" ... "navigated primarily by foot".

**Comment by matkoniecz** [comment]
> I would set it up to yes when not defined. The cyclist would have the possibility to push in the case there is a sign explicitly prohibiting.
> WDYT?

That such routing is a bad idea and if there is no proper data in OSM I would prefer longer trip over chance that I need to dismount

**Comment by matkoniecz** [comment]
@patepelo have you tried adding `bicycle=yes` there if cycling is allowed there?

**Comment by patepelo** [comment]
P.S. This is the query for the Barcelona example, where bicycle != * (empty)

The majority of that is accessible by bicycle.

https://overpass-turbo.eu/s/1K90

**Comment by n1ngu** [comment]
Unless there is a `pedestrian=designated` tag I think all ways should be considered implicitly `bicycle=yes` (except, you know, steps and actual highways).

`highway=pedestrian` can be generally understood as `motor_vehicle=no|destination` but I know no pedestrian way in my whereabouts that forbids bikes access. At some times of the day they may be crowded and riding bike may become slow, but not disallowed. If it was disallowed, then map data is missing `bicycle=no`.

At least the assumption to say "no explicit bicycle tag" --> `bicycle=yes|no` could be left to user decision so one can configure their bike router to use or avoid those ways.

But avoiding them by default seems very **un**reasonable to me @matkoniecz

**Comment by vng** [comment]
To clarify, highway=pedestrian are accessible for bicycles, but the algo weight speed is low (2 km/h by default).
Spain has _NormalPedestrianSpeed_ for bicycles like 5 km/h

See bicycle_model.cpp

Will check this case.

**Comment by vng** [comment]
The algo takes a longer route, because it prefers bicycle=yes ways over cycleway:both=no.
In short, the calculation speed (weight) for bicycle=yes is more than 2x bigger than cycleway:both=no

**Comment by vng** [comment]
No need to add bicycle=yes, unless it is explicitly allowed (by sign for example). Do not map for the router :)

**Comment by AmyMagdalena** [comment]
> No need to add bicycle=yes, unless it is explicitly allowed (by sign for example). Do not map for the router :)

Or disallowed.

**Comment by n1ngu** [comment]
@vng Thanks for looking into it with such detail. I am relieved to know pedestrian streets are taken into account for a bicycle routing and that in this particular case routing through _Carrer de Còrsega_ is a matter of routing weights.

This looks very much like an edge case. All those speed parameters make a lot of sense so this might be a good opportunity to fine tune them.

Fun fact: the whole Barcelona city is notably tilted, sitting in a narrow corridor between a mountain line and a coast line. Not a heavy slope, but local folks develop an instinct to avoid unnecessary NW-SE movements while SW-NE streets are broadly of constant height.

I am explaining this because the current suggested route may look just fine for a flat city but it actually alarmed me because it introduces a significant slope. No cyclist would come up with that route in Barcelona! Yet, we may have never seen this issue if the map was rotated 90º because a similar SW-NE detour would be acceptable to use a faster street.

Is street slope taken into account for bicycle/pedestrian routing? Should be turn this issue into an epic? :sweat_smile:

**Comment by AmyMagdalena** [comment]
> @vng Thanks for looking into it with such detail. I am relieved to know pedestrian streets are taken into account for a bicycle routing and that in this particular case routing through _Carrer de Còrsega_ is a matter of routing weights.
> 
> This looks very much like an edge case. All those speed parameters make a lot of sense so this might be a good opportunity to fine tune them.
> 
> Fun fact: the whole Barcelona city is notably tilted, sitting in a narrow corridor between a mountain line and a coast line. Not a heavy slope, but local folks develop an instinct to avoid unnecessary NW-SE movements while SW-NE streets are broadly of constant height.
> 
> I am explaining this because the current suggested route may look just fine for a flat city but it actually alarmed me because it introduces a significant slope. No cyclist would come up with that route in Barcelona! Yet, we may have never seen this issue if the map was rotated 90º because a similar SW-NE detour would be acceptable so as to use a faster street.
> 
> Is street slope taken into account for bicycle/pedestrian routing? Should we turn this issue into an epic? 😅

This is true of many cities, when I lived in Canterbury, England, to go home, I'd follow the contour of the hill that no router would ever send me along, cause it was about 500m longer and the terrain was terrible, but I'd get home 5m quicker and a lot less exhausted, upwards gradients really need to be weighted heavily against in cycling routing

**Comment by vng** [comment]
@AmyMagdalena Routing takes into account elevation. Could you please share your route on EMail, interesting to check how it works.

**Comment by AmyMagdalena** [comment]
@vng I have sent an email to hello@organicmaps.app about the route in Canterbury, with the recommended organicmaps cycling route, the recommended organicmaps pedestrian route which is close to optimal, and my adjustment to that route that made it speed and energy efficient for cycling, please note I last used this route in the first year of organicmaps's predecessor's existence, it is not a route I can test on the ground for you today as I no longer even live in that country.

**Comment by pastk** [comment]
`highway=pedestrian` is just one of the culprits in this case (this section is quite short after all).

- roads leading to the pedestrian street are `highway=living_street max_speed=10` which is just a tad "faster" than `highway=pedestrian`
- some of the roads chosen by OM have `bicycle=yes` which is over-prioritized by OM https://github.com/organicmaps/organicmaps/issues/9593

**Comment by pastk** [comment]
Also on this intersection https://www.openstreetmap.org/edit#map=22/41.4011024/2.1598300 (and at least some other nearby) footways don't cross the road.

![image](https://github.com/user-attachments/assets/2f16372a-edc2-479c-845e-a2516ac57fa2)

Hence a there-and-back detour in the start on the route seen on Valhalla.

**Comment by matkoniecz** [comment]
opened https://www.openstreetmap.org/note/4502463

**Comment by pastk** [comment]
> opened https://www.openstreetmap.org/note/4502463

The data had updated and it fixed the original route, but moving the starting point slightly shows its an issue still
https://www.openstreetmap.org/directions?engine=fossgis_valhalla_bicycle&route=41.400800%2C2.160260%3B41.399370%2C2.157350

![image](https://github.com/user-attachments/assets/a43dd1b4-45d8-4a07-85f9-2673a556463b)

**Comment by pastk** [comment]
With https://github.com/organicmaps/organicmaps/pull/9692:

![7934_2024-11-19_19-05](https://github.com/user-attachments/assets/c5e20e42-51be-4551-9cb5-8ca29db74c58)
